contracts/src/SuccinctLidoOracleV1.sol

```diff
@@ -73,48 +69,40 @@ contract SuccinctLidoOracleV1 is LidoZKOracle {
         require(REQUESTERS[msg.sender], "only requester can request proof");
         require(!reports[uint256(_refSlot)].requested, "already requested");
 
-        (uint64 availableSlot, bytes32 availableBlockRoot) = findAvailableSlot(_refSlot);
+        bytes32 blockRoot = findBlockRoot(_refSlot);
 
         reports[uint256(_refSlot)].requested = true;
 
         ISuccinctGateway(SUCCINCT_GATEWAY).requestCallback{value: msg.value}(
             FUNCTION_ID,
-            abi.encodePacked(availableBlockRoot, LIDO_WITHDRAWAL_CREDENTIAL),
+            abi.encodePacked(blockRoot, LIDO_WITHDRAWAL_CREDENTIAL),
             abi.encode(_refSlot),
             this.handleUpdate.selector,
             _callbackGasLimit
         );
 
-        emit LidoOracleV1Request(_refSlot, availableSlot, availableBlockRoot);
+        emit LidoOracleV1Request(_refSlot, blockRoot);
     }
 
-    /// @notice Attempts to find the first available slot and its beacon root before or at the given slot.
-    /// @param _refSlot The slot to start searching from.
-    /// @return availableSlot The first available slot found.
-    /// @return availablRoot The beacon root of the first available slot found.
-    /// @dev BEACON_ROOTS returns a block root for a given parent block's timestamp, e.g. to get the block root for slot
-    ///      1000, you use the timestamp of slot 1001.
-    function findAvailableSlot(uint64 _refSlot)
-        public
-        view
-        returns (uint64 availableSlot, bytes32 availablRoot)
-    {
-        uint64 currSlot = _refSlot + 1;
-        bool success;
-        bytes memory result;
-
-        for (uint64 i = 0; i < MAX_SLOT_ATTEMPTS; i++) {
-            if (currSlot == 0) {
-                break;
-            }
-
-            uint256 currTimestamp = GENESIS_BLOCK_TIMESTAMP + (currSlot * 12);
-            (success, result) = BEACON_ROOTS.staticcall(abi.encode(currTimestamp));
+    /// @notice Attempts to find the block root for the given slot.
+    /// @param _slot The slot to get the block root for.
+    /// @return blockRoot The beacon block root of the given slot.
+    /// @dev BEACON_ROOTS returns a block root for a given parent block's timestamp. To get the block root for slot
+    ///      N, you use the timestamp of slot N+1. If N+1 is not avaliable, you use the timestamp of slot N+2, and
+    //       so on.
+    function findBlockRoot(uint64 _slot) public view returns (bytes32 blockRoot) {
+        uint256 currBlockTimestamp = GENESIS_BLOCK_TIMESTAMP + ((_slot + 1) * 12);
```

Comment:
Shouldn't we have a check here that confirms that the `currBlockTimestamp` is within a reasonable distance (i.e. last 8191 slots)? Otherwise, this can return bad data

Reply:
updated